packages/components/src/date-time/time/test/index.tsx

```diff
@@ -89,13 +106,18 @@ describe( 'TimePicker', () => {
 
 		const hoursInput = screen.getByLabelText( 'Hours' );
 
-		fireEvent.change( hoursInput, { target: { value: '22' } } );
-		fireEvent.blur( hoursInput );
+		await user.clear( hoursInput );
+		await user.type( hoursInput, '22' );
+		await user.keyboard( '{Enter}' );
 
-		expect( onChangeSpy ).not.toHaveBeenCalled();
+		expect( onChangeSpy ).toHaveBeenCalledWith( '1986-10-18T00:00:00' );
```

Comment:
Yeah, so previously `TimePicker` rolled its own solution using regular `<input type="number">` elements and was written so that `onChange` would not be called if e.g. `value > max`. We're now using `NumberControl` though which is written so that `onChange` is called with `max` if `value > max`.

I don't really think either behaviour is better or worse, just that it should be consistent across components. I think letting `NumberControl` dictate what happens in this case is the correct approach as that's the point of using standardised components... to make these decisions for us!

Reply:
Sounds good, I agree with your points!